go/kbfs/libkbfs/reporter_kbpki.go

```diff
@@ -271,12 +273,33 @@ func (r *ReporterKBPKI) NotifySyncStatus(ctx context.Context,
 	}
 }
 
+// NotifyOverallSyncStatus implements the Reporter interface for ReporterKBPKI.
+func (r *ReporterKBPKI) NotifyOverallSyncStatus(
+	ctx context.Context, status keybase1.FolderSyncStatus) {
+	select {
+	case r.notifyOverallSyncBuffer <- status:
+	default:
+		// If this represents a "complete" status, we can't drop it.
+		// Instead launch a goroutine to make sure it gets sent
+		// eventually.
+		if status.PrefetchStatus == keybase1.PrefetchStatus_COMPLETE {
+			go func() { r.notifyOverallSyncBuffer <- status }()
```

Comment:
Should we have a shutdown channel here? I'm worried about these goroutines leaking, or panicking racily due to a write to a closed channel when `Shutdown` is called.

Alternatively, what if this buffer was an `InfiniteChannel`? Then it'd never block on write, and we wouldn't have to use this goroutine approach.

Reply:
For what it's worth, this reporter type is not used in tests, only in production, and so I don't think `Shutdown()` will ever be called in a situation where there could be outstanding requests.

But sure, I'll add a shutdown channel just to make it a little cleaner.

Reply:
What about the infinite channel idea?

Reply:
For one, it still has the close race issue.  For another, this particular channel purposely has a size of 1 to avoid unnecessary "incomplete" messages from being sent.  So it would be hard to turn it into an infinite channel while keeping that nice property.  For another another, it's annoying to do the casting away from an `interface{}` with an infinite channel.

Reply:
I see, so you're using the blocking of the channel to filter out incompletes. We could throttle that with a timer instead, but I see the appeal of the current approach.